catalyst/contrib/dl/callbacks/quantization.py

```diff
@@ -0,0 +1,52 @@
+from typing import Dict, Optional, Set, Union
+
+import torch
+from torch import quantization
+
+from catalyst.core import IRunner
+from catalyst.core.callback import Callback, CallbackOrder
+
+
+class DynamicQuantizationCallback(Callback):
+    """
+    Dynamic Quantization Caallback
+
+    This callback applying dynamic quantization
+    of the model at the end of the stage.
+
+    >>> runner.model
+    {"original": ..., "quantized": ...}
+    """
+
+    def __init__(
+        self,
+        qconfig_spec: Optional[Union[Set, Dict]] = None,
+        dtype: Optional[torch.dtype] = torch.qint8,
+    ):
+        """
+        Init method for callback
+        Args:
+            qconfig_spec: torch.quantization.quantize_dynamic
+                parameter, you can define layers to be quantize
+            dtype: type of the model parameters, default int8
+        """
+        super().__init__(CallbackOrder.External)
+        self.qconfig_spec = qconfig_spec
+        self.dtype = dtype
+
+    def on_stage_end(self, runner: "IRunner") -> None:
+        """
+        On stage end action.
+        We are applying quantization here
+        Args:
+            runner: runner of your experiment
+        """
+        quantized_model = quantization.quantize_dynamic(
+            runner.model.cpu(),
+            qconfig_spec=self.qconfig_spec,
+            dtype=self.dtype,
+        )
+        runner.model = {
+            "original": runner.model.cpu(),
+            "quantized": quantized_model,
+        }
```

Comment:
I think we need to parametrise this, to be able to modify model inplace

for example, we could make it in the same logic as https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/dl/callbacks/tracer_callback.py#L11

Reply:
what do you think?